From 2cdedd4939010175bfb093dfa63e6e54e95d3739 Mon Sep 17 00:00:00 2001
From: Keyon Jie <yang.jie@linux.intel.com>
Date: Fri, 13 Jul 2018 13:27:23 +0800
Subject: [PATCH 1334/1670] ASoC: SOF: refine and unify sof_xxx_priv structs

All sof_pci_priv, sof_acpi_priv and sof_spi_priv are exactly same
structs, here unify them and put it to sof-priv.h.

Signed-off-by: Keyon Jie <yang.jie@linux.intel.com>
---
 sound/soc/sof/sof-acpi-dev.c | 11 +++--------
 sound/soc/sof/sof-pci-dev.c  | 11 +++--------
 sound/soc/sof/sof-priv.h     |  9 +++++++++
 sound/soc/sof/sof-spi-dev.c  | 11 +++--------
 4 files changed, 18 insertions(+), 24 deletions(-)

diff --git a/sound/soc/sof/sof-acpi-dev.c b/sound/soc/sof/sof-acpi-dev.c
index fec3b01..9ada41f 100644
--- a/sound/soc/sof/sof-acpi-dev.c
+++ b/sound/soc/sof/sof-acpi-dev.c
@@ -134,14 +134,9 @@ static struct platform_device *
 	return pdev;
 }
 
-struct sof_acpi_priv {
-	struct snd_sof_pdata *sof_pdata;
-	struct platform_device *pdev_pcm;
-};
-
 static void sof_acpi_fw_cb(const struct firmware *fw, void *context)
 {
-	struct sof_acpi_priv *priv = context;
+	struct sof_platform_priv *priv = context;
 	struct snd_sof_pdata *sof_pdata = priv->sof_pdata;
 	const struct snd_soc_acpi_mach *mach = sof_pdata->machine;
 	struct device *dev = &sof_pdata->pdev->dev;
@@ -209,7 +204,7 @@ static int sof_acpi_probe(struct platform_device *pdev)
 	const struct sof_dev_desc *desc;
 	struct snd_soc_acpi_mach *mach;
 	struct snd_sof_pdata *sof_pdata;
-	struct sof_acpi_priv *priv;
+	struct sof_platform_priv *priv;
 	struct snd_sof_dsp_ops *ops;
 	struct platform_device *(*new_mach_data)(struct snd_sof_pdata *pdata);
 	int ret = 0;
@@ -315,7 +310,7 @@ static void sof_acpi_shutdown(struct platform_device *pdev)
 
 static int sof_acpi_remove(struct platform_device *pdev)
 {
-	struct sof_acpi_priv *priv = dev_get_drvdata(&pdev->dev);
+	struct sof_platform_priv *priv = dev_get_drvdata(&pdev->dev);
 	struct snd_sof_pdata *sof_pdata = priv->sof_pdata;
 
 	platform_device_unregister(sof_pdata->pdev_mach);
diff --git a/sound/soc/sof/sof-pci-dev.c b/sound/soc/sof/sof-pci-dev.c
index eaa04ad..470d7d5 100644
--- a/sound/soc/sof/sof-pci-dev.c
+++ b/sound/soc/sof/sof-pci-dev.c
@@ -114,14 +114,9 @@ static struct sof_dev_desc kbl_desc = {
 };
 #endif
 
-struct sof_pci_priv {
-	struct snd_sof_pdata *sof_pdata;
-	struct platform_device *pdev_pcm;
-};
-
 static void sof_pci_fw_cb(const struct firmware *fw, void *context)
 {
-	struct sof_pci_priv *priv = context;
+	struct sof_platform_priv *priv = context;
 	struct snd_sof_pdata *sof_pdata = priv->sof_pdata;
 	const struct snd_soc_acpi_mach *mach = sof_pdata->machine;
 	struct device *dev = sof_pdata->dev;
@@ -192,7 +187,7 @@ static int sof_pci_probe(struct pci_dev *pci,
 		(const struct sof_dev_desc *)pci_id->driver_data;
 	struct snd_soc_acpi_mach *mach;
 	struct snd_sof_pdata *sof_pdata;
-	struct sof_pci_priv *priv;
+	struct sof_platform_priv *priv;
 	struct snd_sof_dsp_ops *ops;
 	int ret = 0;
 
@@ -299,7 +294,7 @@ static void sof_pci_shutdown(struct pci_dev *pci)
 
 static void sof_pci_remove(struct pci_dev *pci)
 {
-	struct sof_pci_priv *priv = pci_get_drvdata(pci);
+	struct sof_platform_priv *priv = pci_get_drvdata(pci);
 	struct snd_sof_pdata *sof_pdata = priv->sof_pdata;
 
 	/* unregister machine driver */
diff --git a/sound/soc/sof/sof-priv.h b/sound/soc/sof/sof-priv.h
index c4162d3..5def46b 100644
--- a/sound/soc/sof/sof-priv.h
+++ b/sound/soc/sof/sof-priv.h
@@ -354,6 +354,15 @@ struct snd_sof_dev {
 };
 
 /*
+ * SOF platform private struct used as drvdata of
+ * platform dev (e.g. pci/acpi/spi...) drvdata.
+ */
+struct sof_platform_priv {
+	struct snd_sof_pdata *sof_pdata;
+	struct platform_device *pdev_pcm;
+};
+
+/*
  * Device Level.
  */
 void snd_sof_shutdown(struct device *dev);
diff --git a/sound/soc/sof/sof-spi-dev.c b/sound/soc/sof/sof-spi-dev.c
index b3811db..45860bf 100644
--- a/sound/soc/sof/sof-spi-dev.c
+++ b/sound/soc/sof/sof-spi-dev.c
@@ -19,14 +19,9 @@
 #include <linux/of_device.h>
 #include "sof-priv.h"
 
-struct sof_spi_priv {
-	struct snd_sof_pdata *sof_pdata;
-	struct platform_device *pdev_pcm;
-};
-
 static void sof_spi_fw_cb(const struct firmware *fw, void *context)
 {
-	struct sof_spi_priv *priv = context;
+	struct sof_platform_priv *priv = context;
 	struct snd_sof_pdata *sof_pdata = priv->sof_pdata;
 	const struct snd_sof_machine *mach = sof_pdata->machine;
 	struct device *dev = sof_pdata->dev;
@@ -61,7 +56,7 @@ static int sof_spi_probe(struct spi_device *spi)
 	const struct snd_sof_machine *mach;
 	struct snd_sof_machine *m;
 	struct snd_sof_pdata *sof_pdata;
-	struct sof_spi_priv *priv;
+	struct sof_platform_priv *priv;
 	int ret = 0;
 
 	dev_dbg(&spi->dev, "SPI DSP detected");
@@ -117,7 +112,7 @@ static int sof_spi_probe(struct spi_device *spi)
 
 static int sof_spi_remove(struct spi_device *spi)
 {
-	struct sof_spi_priv *priv = spi_get_drvdata(spi);
+	struct sof_platform_priv *priv = spi_get_drvdata(spi);
 	struct snd_sof_pdata *sof_pdata = priv->sof_pdata;
 
 	platform_device_unregister(sof_pdata->pdev_mach);
-- 
2.7.4

